packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterInputValue.tsx

```diff
@@ -51,11 +51,6 @@ function GridFilterInputValue(props: GridTypeFilterInputValueProps) {
     [applyValue, item, rootProps.filterDebounceMs, filterTimeout],
   );
 
-  React.useEffect(() => {
-    const itemValue = item.value ?? '';
-    setFilterValueState(String(itemValue));
-  }, [item.value]);
```

Comment:
I believe this effect was covering the use case with a controlled filter model, where the filter model could change while the user has the filter panel open.
Without this effect, the filter input will be out of sync with the actual filter model:

<img width="522" src="https://github.com/mui/mui-x/assets/13808724/061c90c0-df00-4b9e-ade3-26b9403b21f5">

With effect: https://codesandbox.io/s/gracious-khayyam-2xqw3n?file=/demo.tsx
Without effect: https://codesandbox.io/s/dreamy-glitter-h8h4yj?file=/demo.tsx

Reply:
I've adapted the effect to respond to the model changes specifically, what do you think?

Reply:
I'll apply to the other equivalent components if it LGTY.

Reply:
The effect was also making sure the value is correctly synced with header filter, removing it also impacted it too, probably because `fromInput` isn't persisting there?:

Before:
![before-hf](https://github.com/mui/mui-x/assets/12609561/e27d89c5-1093-46c9-b2fc-3820dd843242)

After:
![after-hf](https://github.com/mui/mui-x/assets/12609561/050a69b1-2503-426b-a727-bd791f75904b)

Reply:
Ooh good catch, there's so many parts to this thing ^^

I've pushed a fix for that.

Reply:
Thanks, works as expected now!

Reply:
@romgrk It had another [aspect](https://github.com/mui/mui-x/issues/10325) which I missed catching 😄 
Since the rendering of `GridFilterInputValue` is being done from `GridHeaderFilterCell`, the item change is incorrectly reflected.
I am thinking of disabling this optimization (`fromInput`) for the headerFilters, let me know if you have better ideas.